pkg/systemd/hw-detect.js

```diff
@@ -61,6 +63,60 @@ function findPCI(udevdb, info) {
     }
 }
 
+function findMemoryDevices(udevdb, info) {
+    const memoryArray = [];
+    const dmipath = '/devices/virtual/dmi/id';
+    if (!(dmipath in udevdb)) {
+        return;
+    }
+
+    const props = udevdb[dmipath];
+    // Systemd now exposes memory information in udev, introduced in systemd => 248
+    // https://github.com/systemd/systemd/blob/main/NEWS#L1713
+    if (!('MEMORY_ARRAY_NUM_DEVICES' in props)) {
+        return;
+    }
+
+    for (let slot = 0; slot < props.MEMORY_ARRAY_NUM_DEVICES; slot++) {
+        let memorySize = props[`MEMORY_DEVICE_${slot}_SIZE`];
+        if (memorySize) {
+            memorySize = cockpit.format(_("$0 GiB"), memorySize / 1024 / 1024 / 1024);
+        } else {
+            memorySize = _("Unknown");
+        }
+
+        let memoryRank = props[`MEMORY_DEVICE_${slot}_RANK`] || _("Unknown");
+        if (memoryRank == 1)
+            memoryRank = _("Single rank");
+        if (memoryRank == 2)
+            memoryRank = _("Dual rank");
```

Comment:
Please do this as an `else if` and add an `else` just in case there's ever an "Unknown" or > 2.